### ISSUE TITLE
Laravel `php artisan route:list` fails

### ISSUE THREAD

**Issue by khooz** [GrafiteInc/CMS#129]
Hi,

After I installed the CMS, I tried `php artisan route:list` to inspect available route scopes and names. I faced an error:

```
 Symfony\Component\Debug\Exception\FatalThrowableError  : Type error: Argument 1 passed to Doctrine\Common\Inflector\Inflector::singularize() must be of the type string, null given, called in D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Support\Pluralizer.php on line 84

  at D:\usr\php\laravel\l5\iqfa.local\vendor\doctrine\inflector\lib\Doctrine\Common\Inflector\Inflector.php:441
    437|      * @param string $word The word in plural form.
    438|      *
    439|      * @return string The word in singular form.
    440|      */
  > 441|     public static function singularize(string $word) : string
    442|     {
    443|         if (isset(self::$cache['singularize'][$word])) {
    444|             return self::$cache['singularize'][$word];
    445|         }

  Exception trace:

  1   Doctrine\Common\Inflector\Inflector::singularize()
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Support\Pluralizer.php:84

  2   Illuminate\Support\Pluralizer::singular()
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Support\Str.php:407

  3   Illuminate\Support\Str::singular()
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Support\helpers.php:978

  4   str_singular()
      D:\usr\php\laravel\l5\iqfa.local\vendor\grafite\cms\src\Controllers\ApiController.php:18

  5   Grafite\Cms\Controllers\ApiController::__construct(Object(Illuminate\Http\Request))
      [internal]:0

  6   ReflectionClass::newInstanceArgs()
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Container\Container.php:800

  7   Illuminate\Container\Container::build("Grafite\Cms\Controllers\ApiController")
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Container\Container.php:646

  8   Illuminate\Container\Container::resolve("Grafite\Cms\Controllers\ApiController", [])
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Container\Container.php:601

  9   Illuminate\Container\Container::make("Grafite\Cms\Controllers\ApiController", [])
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Foundation\Application.php:740

  10  Illuminate\Foundation\Application::make("Grafite\Cms\Controllers\ApiController")
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Routing\Route.php:226

  11  Illuminate\Routing\Route::getController()
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Routing\Route.php:796

  12  Illuminate\Routing\Route::controllerMiddleware()
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Routing\Route.php:757

  13  Illuminate\Routing\Route::gatherMiddleware()
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Foundation\Console\RouteListCommand.php:151

  14  Illuminate\Foundation\Console\RouteListCommand::getMiddleware(Object(Illuminate\Routing\Route))
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Foundation\Console\RouteListCommand.php:114

  15  Illuminate\Foundation\Console\RouteListCommand::getRouteInformation(Object(Illuminate\Routing\Route))
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Foundation\Console\RouteListCommand.php:86

  16  Illuminate\Foundation\Console\RouteListCommand::Illuminate\Foundation\Console\{closure}(Object(Illuminate\Routing\Route))
      [internal]:0

  17  array_map(Object(Closure))
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Support\Collection.php:910

  18  Illuminate\Support\Collection::map(Object(Closure))
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Foundation\Console\RouteListCommand.php:87

  19  Illuminate\Foundation\Console\RouteListCommand::getRoutes()
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Foundation\Console\RouteListCommand.php:75

  20  Illuminate\Foundation\Console\RouteListCommand::handle()
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php:29

  21  call_user_func_array([])
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php:29

  22  Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php:87

  23  Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Object(Closure))
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php:31

  24  Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), [])
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Container\Container.php:564

  25  Illuminate\Container\Container::call()
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Console\Command.php:183

  26  Illuminate\Console\Command::execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
      D:\usr\php\laravel\l5\iqfa.local\vendor\symfony\console\Command\Command.php:252

  27  Symfony\Component\Console\Command\Command::run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Console\Command.php:170

  28  Illuminate\Console\Command::run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      D:\usr\php\laravel\l5\iqfa.local\vendor\symfony\console\Application.php:865

  29  Symfony\Component\Console\Application::doRunCommand(Object(Illuminate\Foundation\Console\RouteListCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      D:\usr\php\laravel\l5\iqfa.local\vendor\symfony\console\Application.php:241

  30  Symfony\Component\Console\Application::doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      D:\usr\php\laravel\l5\iqfa.local\vendor\symfony\console\Application.php:143

  31  Symfony\Component\Console\Application::run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Console\Application.php:88

  32  Illuminate\Console\Application::run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      D:\usr\php\laravel\l5\iqfa.local\vendor\laravel\framework\src\Illuminate\Foundation\Console\Kernel.php:122

  33  Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      D:\usr\php\laravel\l5\iqfa.local\artisan:37
```

It roots down to getting some middleware information from `Grafite\Cms\Controllers\ApiController` constructor and pass it to `Pluralizer::singular()` with a `null` argument instead of an `string`.

**Update**

The culprit is the part below:

https://github.com/GrafiteInc/CMS/blob/e488e4819237b24fd515f81d866348307aa9bc8e/src/Controllers/ApiController.php#L18

I could just fix the `route:list` problem by null-checking the 3rd segment

```php
$this->modelName = str_singular($request->segment(3) ?? '');
```

I don't know how this affects the rest of the application yet, so I just use the fix whenever I need `route:list`.

Cheers!

**Comment by mlantz** [comment]
Hey I found the root of this and the fix is coming in the next release

**Comment by mlantz** [comment]
Resolved in v3.1
